src/plugins/workspace/public/components/workspace_menu/workspace_menu.tsx

```diff
@@ -0,0 +1,191 @@
+        </EuiText>
+      ) : (
+        workspace.name
+      );
+    return {
+      name,
+      key: workspace.id,
+      icon: <EuiIcon type="stopFilled" color={workspace.color ?? 'primary'} />,
+      onClick: () => {
+        window.location.assign(workspaceURL);
+      },
+    };
+  };
+
+  const getWorkspaceListItems = () => {
+    const workspaceListItems: EuiContextMenuPanelItemDescriptor[] = filteredWorkspaceList.map(
+      workspaceToItem
+    );
+    workspaceListItems.push({
+      icon: <EuiIcon type="plus" />,
+      name: i18n.translate('core.ui.primaryNav.workspaceContextMenu.createWorkspace', {
+        defaultMessage: 'Create workspace',
+      }),
+      key: WORKSPACE_CREATE_APP_ID,
+      onClick: () => {
+        window.location.assign(
+          cleanWorkspaceId(
+            coreStart.application.getUrlForApp(WORKSPACE_CREATE_APP_ID, {
+              absolute: false,
+            })
+          )
+        );
+      },
+    });
+    workspaceListItems.push({
+      icon: <EuiIcon type="folderClosed" />,
+      name: i18n.translate('core.ui.primaryNav.workspaceContextMenu.allWorkspace', {
+        defaultMessage: 'All workspaces',
+      }),
+      key: WORKSPACE_LIST_APP_ID,
+      onClick: () => {
+        window.location.assign(
+          cleanWorkspaceId(
+            coreStart.application.getUrlForApp(WORKSPACE_LIST_APP_ID, {
+              absolute: false,
+            })
+          )
+        );
+      },
+    });
+    return workspaceListItems;
+  };
+
+  const currentWorkspaceButton = (
+    <>
+      <EuiListGroup style={{ width: 318 }} maxWidth={false}>
+        <EuiListGroupItem
+          iconType="spacesApp"
+          label={currentWorkspaceName}
+          onClick={openPopover}
+          extraAction={{
+            color: 'subdued',
+            onClick: openPopover,
+            iconType: isPopoverOpen ? 'arrowDown' : 'arrowRight',
+            iconSize: 's',
+            'aria-label': 'Show workspace dropdown selector',
+            alwaysShow: true,
+          }}
+        />
+      </EuiListGroup>
+    </>
+  );
+
+  const currentWorkspaceTitle = (
+    <EuiFlexGroup alignItems="center">
+      <EuiFlexItem grow={true}>
+        <EuiText size="s">{currentWorkspaceName}</EuiText>
+      </EuiFlexItem>
+      <EuiFlexItem grow={false}>
+        <EuiButtonIcon
+          iconType="cross"
+          onClick={closePopover}
+          aria-label="close workspace dropdown"
+        />
+      </EuiFlexItem>
+    </EuiFlexGroup>
+  );
+
+  const panels = [
+    {
+      id: 0,
+      title: currentWorkspaceTitle,
+      items: getWorkspaceListItems(),
+    },
+  ];
+
+  return (
+    <EuiPopover
+      id="workspaceDropdownMenu"
+      display="block"
+      button={currentWorkspaceButton}
+      isOpen={isPopoverOpen}
+      closePopover={closePopover}
+      panelPaddingSize="none"
+      anchorPosition="downCenter"
+    >
+      <EuiContextMenu initialPanelId={0} panels={panels} />
```

Comment:
Thanks for this change. 
One question:

The id here is 0 https://github.com/opensearch-project/OpenSearch-Dashboards/pull/6150/files#diff-907ad2efa095285762230d1f2131f64d24c6f03e80f2a5a1ea4122a73a4cf20bR172

Also we set initialPanelId={0} here.

Will it be a duplicate?

Thanks

Reply:
No, I think the `initialPanelId` defines the initial panel when the context menu popup, the id should match what we defined in `panels`.